homework-submission/week2-homework/src/App.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import "./App.css";
+import TodoItem from "./components/TodoItem";
+import List from "./components/List";
+import Checkbox from "./components/Checkbox";
+import todos from "./source/todos.json";
+import Header from "./components/Header";
+import image from "./todo.jpg";
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(id) {
+    let todos = this.state.todos.slice();
+    let foundTodo = todos.find(todo => id === todo.id);
+    foundTodo.done = !foundTodo.done;
+    this.setState({ todos: todos });
```

Comment:
I used your version in my application but it does not change `todo.done` state in a right way. I console.logged it and Instead of `true`, I got `false` and instead of  `false` I got true. Can you maybe clarify this? Maybe I am doing something wrong?

Reply:
Hmm, That's wired I will try it again and get back!

Reply:
![ezgif-2-f4bd4ba380](https://user-images.githubusercontent.com/25272822/45255988-21e33100-b390-11e8-9ec3-79844e10d0db.gif)

Reply:
The last thing, Very good question.

I won't cheat you because I have no real Idea about why 🙃, But I know it doing that.
I think the slice here is just creating a new reference for the original state, and we should ask about it on Sunday what do you say? 😬

but as you see it's doing it right?